crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,242 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::{
+    runtime_api::api, state_call, submit_extrinsic, Balance, Client, CodeHash,
+    ContractMessageTranscoder, CrateMetadata, DefaultConfig, ExtrinsicOpts, PairSigner,
+    TokenMetadata,
+};
+use crate::{
+    cmd::extrinsics::{events::DisplayEvents, ErrorVariant},
+    name_value_println,
+    util::decode_hex,
+};
+use anyhow::{anyhow, Context, Result};
+use pallet_contracts_primitives::CodeUploadResult;
+use scale::Encode;
+use std::{
+    fmt::Debug,
+    fs,
+    path::{Path, PathBuf},
+};
+use subxt::{Config, OnlineClient};
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "remove", about = "Remove a contract's code")]
+pub struct RemoveCommand {
+    /// Path to Wasm contract code, defaults to `./target/ink/<name>.wasm`.
+    #[clap(value_parser)]
+    wasm_path: Option<PathBuf>,
+    /// The hash of the smart contract code already uploaded to the chain.
+    /// If the contract has not already been uploaded use `--wasm-path` or run the `upload` command
+    /// first.
+    #[clap(long, value_parser = parse_code_hash)]
+    code_hash: Option<<DefaultConfig as Config>::Hash>,
+    #[clap(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+/// Parse a hex encoded 32 byte hash. Returns error if not exactly 32 bytes.
+fn parse_code_hash(input: &str) -> Result<<DefaultConfig as Config>::Hash> {
+    let bytes = decode_hex(input)?;
+    if bytes.len() != 32 {
+        anyhow::bail!("Code hash should be 32 bytes in length")
+    }
+    let mut arr = [0u8; 32];
+    arr.copy_from_slice(&bytes);
+    Ok(arr.into())
+}
+
+impl RemoveCommand {
+    pub fn is_json(&self) -> bool {
+        self.output_json
+    }
+
+    pub fn run(&self) -> Result<(), ErrorVariant> {
+        let crate_metadata = CrateMetadata::from_manifest_path(
+            self.extrinsic_opts.manifest_path.as_ref(),
+        )?;
+        let contract_metadata =
+            contract_metadata::ContractMetadata::load(&crate_metadata.metadata_path())?;
+        let transcoder = ContractMessageTranscoder::load(crate_metadata.metadata_path())?;
+        let signer = super::pair_signer(self.extrinsic_opts.signer()?);
+
+        fn load_code(wasm_path: &Path) -> Result<Code> {
+            tracing::debug!("Contract code path: {}", wasm_path.display());
+            let code = fs::read(wasm_path)
+                .context(format!("Failed to read from {}", wasm_path.display()))?;
+            Ok(Code::Upload(code))
+        }
+
+        let code = match (self.wasm_path.as_ref(), self.code_hash.as_ref()) {
```

Comment:
I think we get rid of `wasm-path` and use (in order or precedence): `self.code_hash`, otherwise `contract_metadata.source.hash`

Reply:
See #893